### ISSUE TITLE
selecting a button in one month and then changing the month to a month where that same button becomes a special date without an explicit border automatically gives it the selected border style

### ISSUE THREAD

**Issue by KRA2008** [rebeccaXam/XamForms.Controls.Calendar#16]
1) In code, make Jan 1 2017 a special date
2) In the app, navigate to December
2) Tap/select Nov 27
3) Navigate to January
4) Jan 1 will be highlighted as a special date and as though it were selected

I'm going to come back and take a look at this soon.

**Comment by rebeccaXam** [comment]
Thanks, it could be a race condition. I will release a new version (1.0.6) at the end of this week would you check this again with that version. 
I could not reproduce this, but I already changed some code, so I do not know if this problem was solve by that.

**Comment by KRA2008** [comment]
It seems to only happen when the SpecialDate doesn't specify an explicit border color. In the example app this would be the SpecialDate set 3 days after today. I'm guessing when the month is changed the button just keeps the border color the same as it was when it was in the previous month because the incoming SpecialDate doesn't specify anything to set it to. It looks like the analogous problem happens for SelectedFontSize.

At first blush I might suggest setting all buttons to normal when the month changes before going back and layering special & selected stylings on top. That might mean setting a few colors and things twice (maybe inefficient) but it would make it easier to maintain the "default" styling.

**Comment by KRA2008** [comment]
Looking closer, the suggestion I made might work, but it looks like you already have some code for handling selected buttons that does the sort of handling of default styles we would need for SpecialDates:
```
				button.BackgroundColor = SelectedBackgroundColor.HasValue ? SelectedBackgroundColor.Value : (special != null && special.BackgroundColor.HasValue ? special.BackgroundColor.Value : defaultBackgroundColor);
				button.TextColor = SelectedTextColor.HasValue ? SelectedTextColor.Value : (special != null && special.TextColor.HasValue ? special.TextColor.Value : defaultTextColor);
```
I guess one of the approaches would need to be chosen but I'm not sure of the full consequences of the each yet.

**Comment by rebeccaXam** [comment]
I hoop this is resolved in 1.0.8, if not please reopen this issue.
